kv/txn_correctness_test.go

```diff
@@ -145,78 +152,68 @@ func (c *cmd) String() string {
-	if err := db.Run(call); err != nil {
+	r, err := db.Get(c.getKey())
+	if err != nil {
 		return err
 	}
 	if r.Value != nil {
-		c.env[c.key] = r.Value.GetInteger()
-		c.debug = fmt.Sprintf("[%d ts=%d]", r.Value.GetInteger(), r.Timestamp.Logical)
+		c.env[c.key] = r.ValueInt()
+		c.debug = fmt.Sprintf("[%d ts=%d]", r.ValueInt(), r.Timestamp)
 	}
 	return nil
 }
 
 // deleteRngCmd deletes the range of values from the db from [key, endKey).
 func deleteRngCmd(c *cmd, db runner, t *testing.T) error {
-	return db.Run(client.DeleteRange(c.getKey(), c.getEndKey()))
+	return db.DelRange(c.getKey(), c.getEndKey())
 }
 
 // scanCmd reads the values from the db from [key, endKey).
 func scanCmd(c *cmd, db runner, t *testing.T) error {
-	call := client.Scan(c.getKey(), c.getEndKey(), 0)
-	r := call.Reply.(*proto.ScanResponse)
-	if err := db.Run(call); err != nil {
+	rows, err := db.Scan(c.getKey(), c.getEndKey(), 0)
+	if err != nil {
 		return err
 	}
 	var vals []string
 	keyPrefix := []byte(fmt.Sprintf("%d.", c.historyIdx))
-	for _, kv := range r.Rows {
+	for _, kv := range rows {
 		key := bytes.TrimPrefix(kv.Key, keyPrefix)
-		c.env[string(key)] = kv.Value.GetInteger()
-		vals = append(vals, fmt.Sprintf("%d", kv.Value.GetInteger()))
+		c.env[string(key)] = kv.ValueInt()
+		vals = append(vals, fmt.Sprintf("%d", kv.ValueInt()))
 	}
-	c.debug = fmt.Sprintf("[%s ts=%d]", strings.Join(vals, " "), r.Timestamp.Logical)
+	c.debug = fmt.Sprintf("[%s]", strings.Join(vals, " "))
 	return nil
 }
 
 // incCmd adds one to the value of c.key in the env and writes
 // it to the db. If c.key isn't in the db, writes 1.
 func incCmd(c *cmd, db runner, t *testing.T) error {
-	call := client.Increment(c.getKey(), 1)
-	r := call.Reply.(*proto.IncrementResponse)
-	if err := db.Run(call); err != nil {
+	r, err := db.Inc(c.getKey(), 1)
+	if err != nil {
 		return err
 	}
-	c.env[c.key] = r.NewValue
-	c.debug = fmt.Sprintf("[%d ts=%d]", r.NewValue, r.Timestamp.Logical)
+	c.env[c.key] = r.ValueInt()
+	c.debug = fmt.Sprintf("[%d]", r.ValueInt())
 	return nil
 }
 
-// sumCmd sums the values of all keys read during the transaction
-// and writes the result to the db.
+// sumCmd sums the values of all keys != c.key read during the transaction and
+// writes the result to the db.
 func sumCmd(c *cmd, db runner, t *testing.T) error {
 	sum := int64(0)
-	for _, v := range c.env {
-		sum += v
-	}
-	r := &proto.PutResponse{}
-	err := db.Run(client.Call{
-		Args: &proto.PutRequest{
-			RequestHeader: proto.RequestHeader{Key: c.getKey()},
-			Value:         proto.Value{Integer: gogoproto.Int64(sum)},
-		},
-		Reply: r})
-	c.debug = fmt.Sprintf("[%d ts=%d]", sum, r.Timestamp.Logical)
+	for k, v := range c.env {
+		if k != c.key {
+			sum += v
+		}
+	}
+	r, err := db.Inc(c.getKey(), sum)
+	c.debug = fmt.Sprintf("[%d ts=%d]", sum, r.Timestamp)
 	return err
 }
 
 // commitCmd commits the transaction.
 func commitCmd(c *cmd, db runner, t *testing.T) error {
```

Comment:
Yeah, that's what I'd like to do. Not easy because `commitCmd` needs to have the same signature as the other `*Cmd` functions so that it can be placed in `cmdDict`. And those commands cannot all take a `txRunner` because they are called with `*client.DB` during history verification.